plugin/cql/src/main/java/org/opencds/cqf/ruler/cql/KnowledgeArtifactAdapter.java

```diff
@@ -99,6 +129,30 @@ public List<RelatedArtifact> getRelatedArtifactsByType(RelatedArtifact.RelatedAr
 		return relatedArtifacts;
 	}
 
+	public List<RelatedArtifact> getOwnedRelatedArtifacts(){
+		switch (resource.getClass().getSimpleName()) {
+			case "ActivityDefinition":
+			case "Library":
+			case "Measure":
+			case "PlanDefinition":
+				return getOwnedRelatedArtifactsOfKnowledgeArtifact();
+			default :
+				return new ArrayList<>();
+		}
+	}
+	private List<RelatedArtifact> getOwnedRelatedArtifactsOfKnowledgeArtifact() {
+		return getRelatedArtifact().stream()
+			.filter(ra -> checkIfRelatedArtifactIsOwned(ra))
+			.collect(Collectors.toList());
+	}
+	static Boolean checkIfRelatedArtifactIsOwned(RelatedArtifact ra){
+		return ra.getExtension()
+					.stream()
+					.filter(ext -> ext.getUrl().equals("http://hl7.org/fhir/StructureDefinition/crmi-isOwned"))
+					.findAny()
+					.map(e -> ((BooleanType) e.getValue()).getValue())
+					.orElseGet(()-> false);
+	}
```

Comment:
I made this a static method since it's a pure function which doesn't need an instance of the adapter, but does this mean it should be moved somewhere else and imported as a utility?

Reply:
I think this is fine. When we refactor down into the Evaluator stack we can consider this and whether or not there is something in place there that we should respect and leverage (i.e., a util class).